python/cudf/cudf/core/single_column_frame.py

```diff
@@ -106,7 +107,62 @@ def _column(self) -> ColumnBase:
     @property  # type: ignore
     @_performance_tracking
     def values(self) -> cupy.ndarray:
-        return self._column.values
+        col = self._column
+        if (
+            self.ndim == 1
+            and not col.has_nulls()
+            and col.dtype.kind in {"i", "u", "f", "b"}
```

Comment:
```suggestion
            col.dtype.kind in {"i", "u", "f", "b"}
            and not col.has_nulls()
```

Nit:

1. `self.ndim == 1` is always true for this class
2. The dtype check is essentially always cheaper than the `has_null` check since `null_count` might not be computed yet.